ndc-models/src/lib.rs

```diff
@@ -691,9 +832,23 @@ pub struct RowSet {
     pub aggregates: Option<IndexMap<String, serde_json::Value>>,
     /// The rows returned by the query, corresponding to the query's fields
     pub rows: Option<Vec<IndexMap<String, RowFieldValue>>>,
+    /// The results of any grouping operation applies to the returned rows
+    pub groups: Option<Vec<Group>>,
 }
 // ANCHOR_END: RowSet
 
+// ANCHOR: Group
+#[skip_serializing_none]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Group")]
+pub struct Group {
+    /// Values of dimensions which identify this group
+    pub dimensions: Vec<serde_json::Value>,
```

Comment:
Why is this not an `IndexMap<String, serde_json::Value>`?

Reply:
This is how it was before, but @daniel-chambers made the good point that the ordering of dimensions may be significant, and JSON at the level of values does not respect the ordering of fields.